src/sentry_core.c

```diff
@@ -457,9 +459,14 @@ sentry__should_send_transaction(sentry_value_t tx_cxt)
 
     bool send = false;
     SENTRY_WITH_OPTIONS (options) {
-        send = sentry__roll_dice(options->traces_sample_rate);
-        // TODO(tracing): Run through traces sampler function if rate is
-        // unavailable.
+        if (options->traces_sampler) {
+            double result
+                = ((double (*)(void *))options->traces_sampler)(sampling_ctx);
+            send = sentry__roll_dice(result);
+        } else {
+            // TODO if there is a parent sampling decision, use it
```

Comment:
I think we already do? In `transaction_context_update_from_header_n` we [get the parent sampling decision](https://github.com/getsentry/sentry-native/blob/master/src/sentry_tracing.c#L262) if it exists, and in this `should_send_transaction` we [check if the `tx_cxt` has this key already set ](https://github.com/getsentry/sentry-native/blob/master/src/sentry_core.c#L453-L456)(which, if the parent had a sampling decision, will be the case).

Reply:
+ following the develop docs section on [precedence](https://docs.sentry.io/platforms/native/configuration/sampling/#precedence) the order should be `passed in decision` (not supported) > `traces_sampler` > parent decision > `traces_sample_rate`, so if I understand the above fetching of the `"sampled"` key correctly, we should change the order.

Reply:
Yes to all adaptations. 😄